src/PJ_helmert.c

```diff
@@ -520,12 +520,15 @@ PJ *PROJECTION(helmert) {
     if (pj_param (P->ctx, P->params, "ttheta").i) {
         Q->theta_0 = pj_param (P->ctx, P->params, "dtheta").f * ARCSEC_TO_RAD;
         Q->fourparam = 1;
+        Q->scale_0 = 1.0; /* default scale for the 4-param shift */
     }
 
     /* Scale */
-    if (pj_param (P->ctx, P->params, "ts").i)
+    if (pj_param (P->ctx, P->params, "ts").i) {
         Q->scale_0 = pj_param (P->ctx, P->params, "ds").f;
-
+        if (pj_param (P->ctx, P->params, "ttheta").i && Q->scale_0 == 0.0)
+            return freeup_msg(P, -52);
```

Comment:
I hate this old habit of hardcoding error codes. Shouldn't we extend the list in project.h with a #define PJD_ERR_SCALE_CANNOT_BE_ZERO -52 ? and in the comment of the new line in pj_strerrno, mention this symbolic name as well

In that case, shouldn't we actually generalize the error to PJD_INVALID_SCALE if negative or other values are found to be non sensical at some point ?